tools/ci/actions/check-changes-modified.sh

```diff
@@ -20,14 +20,24 @@ set -e
 #   - A commit in the PR alters the Changes file
 #   - The no-change-entry-needed label is applied to the PR (handled in YAML)
 
+API_URL="$1"
+shift 1
+
+AUTH="authorization: Bearer $GITHUB_TOKEN"
+
 # We need all the commits in the PR to be available
 . tools/ci/actions/deepen-fetch.sh
 
 MSG='Check Changes has been updated'
 COMMIT_RANGE="$MERGE_BASE..$PR_HEAD"
 
 # Check if Changes has been updated in the PR
-if git diff "$COMMIT_RANGE" --name-only --exit-code Changes > /dev/null; then
+if ! git diff "$COMMIT_RANGE" --name-only --exit-code Changes > /dev/null; then
+  echo -e "$MSG: \e[32mYES\e[0m"
+elif [[ -n $(curl --silent --header "$AUTH" "$API_URL/labels" | \
+            grep 'no-change-entry-needed') ]]; then
```

Comment:
Looks like black magic, but if you're confident...

Reply:
It's a combination of a rune from the [GitHub Actions docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-2-calling-the-rest-api) and the original rune (due to @gasche) from the old Travis scripts, seasoned with a little bashism 🙂

Reply:
What about
```
elif curl .... | grep -q 'no-change-entry-needed'; then
```
(No need to store the output of grep, then check that it's nonempty.)

Reply:
Indeed - amended